File: examples/scripts/demo1.js
// Query program version.
print("This is Ovito " + ovito.version)

// Import a data file.
node = load("../data/NanocrystallinePd.dump.gz")

// Block execution of the script until the scene is ready, that is, until 
// the input file has been completely loaded.
// This is an optional step, but it ensures that the modifier we are going to create 
// has access to the input data at the time it is inserted into the modification pipeline.
// This allows the Color Coding modifier to automatically adjust its interval to the range of 
// values present in the input data.
wait()

// Apply a modifier to the dataset.
node.applyModifier(new ColorCodingModifier({ 
	sourceProperty : "Potential Energy",
	colorGradient  : new ColorCodingHotGradient()
}))

// Set up view, looking along the [2,3,-3] vector.
activeViewport.perspective(Point(-100, -150, 150), Vector(2, 3, -3), 60.0 * Math.PI/180.0)

// Render a picture of the dataset.
activeViewport.render({
	filename    : "rendering.png",
	imageWidth  : 120,
	imageHeight : 120
})

// Apply two more modifiers to delete some particles.
node.applyModifier(new SelectExpressionModifier({ expression : "PotentialEnergy < -3.9" }))
node.applyModifier(new DeleteParticlesModifier())

// Print the modification pipeline of the selected node to the console.
print("Current modification pipeline:")
for(var i = 0; i < node.modifiers.length; i++)
	print("  " + node.modifiers[i])  
	
// Perform some analysis.
cna = new CommonNeighborAnalysisModifier({ cutoff : 3.2, adaptiveMode : false })
node.applyModifier(cna)

// Wait until computation has been completed.
wait()

// Read out analysis results.
print("Number of FCC atoms: " + cna.structureCounts[CommonNeighborAnalysisModifier.FCC])

// Write processed atoms back to an output file.
save("exporteddata.dump", LAMMPSDumpExporter, 
	{ columnMapping: ["Position.X", "Position.Y", "Position.Z", "Structure Type"] })

File: examples/scripts/demo1.py
# Import OVITO modules.
from ovito import *
from ovito.io import *
from ovito.modifiers import *
from ovito.vis import *
from ovito.data import *

# Import modules from standard Python library.
import math
import numpy as np

# Query program version.
print "This is Ovito version", version

# Import a data file.
node = import_file("../data/NanocrystallinePd.dump.gz")

# Apply a modifier to the dataset.
node.modifiers.append(ColorCodingModifier(
	property = ParticleProperty.Type.PotentialEnergy,
	gradient = ColorCodingModifier.Hot()
))

# Set up view, looking along the [2,3,-3] direction from camera position (-100, -150, 150).
vp = Viewport()
vp.type = Viewport.Type.PERSPECTIVE
vp.camera_pos = (-100, -150, 150)
vp.camera_dir = (2, 3, -3)
vp.fov = math.radians(60.0)

settings = RenderSettings(
	filename = "rendered_image.png",
	size = (220,220)
)

# Render a picture of the dataset.
vp.render(settings)

# Apply two more modifiers to delete some particles.
node.modifiers.append(SelectExpressionModifier(expression = "PotentialEnergy < -3.9"))
node.modifiers.append(DeleteSelectedParticlesModifier())

# Print the modification pipeline of the selected node to the console.
print "Modification pipeline:"
print node.modifiers
	
# Perform some analysis.
cna = CommonNeighborAnalysisModifier(adaptive_mode = True)
node.modifiers.append(cna)

# Block execution of the script until the node's modification pipeline is ready, that is, until 
# the results of all modifiers have been computed.
node.compute()
for c in enumerate(cna.counts):
	print "Type %i: %i particles" % c
print node.output['Position'].array
for t in node.output.structure_type.type_list:
	print t.id, t.name, t.color

# Read out analysis results.
print "Number of FCC atoms:", cna.counts[CommonNeighborAnalysisModifier.Type.FCC]

# Write processed atoms back to an output file.
export_file(node, "exported_data.dump", "lammps_dump", 	
	columns = ["Position.X", "Position.Y", "Position.Z", "Structure Type"]
)
